src/Native/Unix/System.Native/pal_io.c

```diff
@@ -995,14 +995,21 @@ int32_t SystemNative_ReadLink(const char* path, char* buffer, int32_t bufferSize
     assert(buffer != NULL || bufferSize == 0);
     assert(bufferSize >= 0);
 
-    if (bufferSize < 0)
+    if (bufferSize <= 0)
     {
         errno = EINVAL;
         return -1;
     }
 
     ssize_t count = readlink(path, buffer, (size_t)bufferSize);
     assert(count >= -1 && count <= bufferSize);
+
+    // Check if buffer may be truncated.
+    if (count == bufferSize)
+    {
+        return 0;
```

Comment:
I'm not sure if I like that we are adding semantics to our native shims.  It makes it hard to correlate our `SystemNative_XXX` behavior with the man pages for the corresponding function.

Would it be OK if this was handled in managed code instead?